### DISCUSSION TITLE
Error when persisting store with nested "actions" object using local storage

### DISCUSSION THREAD

**Discussion by soonbenjy** [pmndrs/zustand#1561] [Q&A] [answered]
Hello,

I have an interface for a store that contains a state and actions in a separate nested object:

```
interface BannerStore {
  bannersToHide: BannerKey[]
  actions: {
    addBannerToHide: (bannerKey: BannerKey) => void
  }
}
```

For a new feature, I decided to persist the entire store using local storage. However, when persisting the "actions" property as an empty object (since functions cannot be persisted), I am getting the error "addBannerToHide is not a function" when returning to the application with the local storage pre-populated.

A workaround for this issue would be to omit the "actions" property when persisting the store, using the "partialize" function.

Is it possible to not rely on local storage for the store's actions, even if they are nested in an "actions" object (instead of being flattened)?

**Comment by dai-shi** [comment, marked as answer]
How about defining actions outside?

```js
const useBannerStore = create(() => ({
  bannersToHide: [],
}))

const actions = {
  addBannerToHide: (bannerKey) => {
    const { getState, setState } = useBannerStore
    // ...
  },
}
```

Related: https://github.com/pmndrs/zustand/blob/2f2fee4803044b568f5fc1842e106ec68a770d2d/docs/guides/practice-with-no-store-actions.md
